x-pack/filebeat/input/httpinput/input.go

```diff
@@ -0,0 +1,329 @@
+	if err != "" || status != 0 {
+		return status, err
+	}
+
+	// Create the event
+	ok := in.outlet.OnEvent(beat.Event{
+		Timestamp: time.Now().UTC(),
+		Fields: common.MapStr{
+			"message":        "testing",
+			in.config.Prefix: in.eventObject,
+		},
+	})
+
+	// If event cannot be sent
+	if !ok {
+		return http.StatusInternalServerError, in.createErrorMessage("unable to send event")
+	}
+
+	return 0, ""
+}
+
+func (in *HttpInput) validateRequest() (uint, string) {
+	// Only allow POST requests
+	var err string
+	var status uint
+
+	// Check auth settings and credentials
+	if in.config.BasicAuth == true {
+		status, err = in.validateAuth()
+	}
+
+	if err != "" && status != 0 {
+		return status, err
+	}
+
+	// Validate headers
+	status, err = in.validateHeader()
+
+	if err != "" && status != 0 {
+		return status, err
+	}
+
+	// Validate body
+	status, err = in.validateBody()
+
+	if err != "" && status != 0 {
+		return status, err
+	}
+
+	return 0, ""
+
+}
+
+func (in *HttpInput) validateHeader() (uint, string) {
+	// Only allow JSON
+	if in.httpRequest.Header.Get("Content-Type") != "application/json" {
+		return http.StatusUnsupportedMediaType, in.createErrorMessage("wrong content-type header, expecting application/json")
+	}
+
+	// Only accept JSON in return
+	if in.httpRequest.Header.Get("Accept") != "application/json" {
+		return http.StatusNotAcceptable, in.createErrorMessage("wrong accept header, expecting application/json")
+	}
+	return 0, ""
+}
+
+func (in *HttpInput) validateAuth() (uint, string) {
+	// Check if username or password is missing
+	if in.config.Username == "" || in.config.Password == "" {
+		return http.StatusUnauthorized, in.createErrorMessage("Username and password required when basicauth is enabled")
+	}
+
+	// Check if username and password combination is correct
+	username, password, _ := in.httpRequest.BasicAuth()
+	if in.config.Username != username || in.config.Password != password {
+		return http.StatusUnauthorized, in.createErrorMessage("Incorrect username or password")
+	}
+
+	return 0, ""
+}
+
+func (in *HttpInput) validateBody() (uint, string) {
+	// Checks if body is empty
+	if in.httpRequest.Body == http.NoBody {
+		return http.StatusNotAcceptable, in.createErrorMessage("body can not be empty")
+	}
+
+	// Write full []byte to string
+	body, err := ioutil.ReadAll(in.httpRequest.Body)
+
+	// If body cannot be read
+	if err != nil {
+		return http.StatusInternalServerError, in.createErrorMessage("unable to read body")
+	}
+
+	// Declare interface for request body
+	objmap := make(map[string]interface{})
+
+	err = json.Unmarshal(body, &objmap)
+
+	// If body can be read, but not converted to JSON
+	if err != nil {
+		return http.StatusBadRequest, in.createErrorMessage("malformed JSON body")
+	}
+	// Assign the current Unmarshaled object when no errors
+	in.eventObject = &objmap
+
+	return 0, ""
+}
+
+func (in *HttpInput) validateMethod() (uint, string) {
+	// Ensure HTTP method is POST
```

Comment:
This comment seems to me like a "header" comment which should go above the function declaration. E.g.
```golang
// validateMethod ensures that the HTTP method is POST
func (in *HttpInput) validateMethod() (uint, string) {
```